### DISCUSSION TITLE
[Question]: Is it possible to import calendar/addressbook

### DISCUSSION THREAD

**Discussion by mbauhardt** [jelmer/xandikos#152] [Q&A] [answered]
Hi,
i have a question if it is possible to import a calendar or addressbook from another system.
my use case is i want to switch from radicale to xandikos. but i have several calendar events and addressbook entries.
do you think to copy all *.ics or *.vcf files into a collection owned by xandikos and then the events and contacts are available. using git commit to commit the files into the repo.
or do you see another way?

**Comment by legrostdg** [comment]
You can use `vdirsyncer` for this.

**Comment by jelmer** [comment, marked as answer]
Yeah, you can just commit the vcf and ics files to the repository directly. There is nothing special about the repository, it's just like any other git repository.

If you have both the radicale and the xandikos server running at the same time, then you can also use a WebDAV/caldav/carddav client to copy over the items. E.g. vdirsyncer, as legrostdg suggests.

**Reply by bryankaplan** [reply]
@jelmer, could you please document how the former approach works? Where have I gone wrong?
1. Initial server setup:
```
./venv/bin/xandikos --defaults --no-detect-systemd \
    --directory="$XDG_DATA_HOME/xandikos" --current-user-principal="/$USER"
```
2. I exported an ics file from Google Calendar.
3. I put the ics file in `$XDG_DATA_HOME/xandikos/$USER/calendars/calendar/`.
4. I did `git add` and `git commit`.
5. `cadaver http://localhost:8080/$USER/calendars/calendar`
6. `dav:/[REDACTED]/calendars/calendar/> ls` produces:
```
Listing collection `/[REDACTED]/calendars/calendar/': succeeded.
Error: calendar.ics                   404 Not Found
```
Additionally, just to make sure, `dav:/[REDACTED]/calendars/calendar/> ls calendar.ics` produces:
``` 
Listing collection `/[REDACTED]/calendars/calendar/calendar.ics/': collection is empty.
```
Incidentally, the file is 2 MB.

**Comment by mbauhardt** [comment]
hey,
thx @legrostdg @jelmer 

clever/logic idea to use vdirsyncer to sync the contact from a to b.
will do it.
